assets/app/scripts/services/projects.js

```diff
@@ -1,19 +1,30 @@
 'use strict';
 
 angular.module('openshiftConsole')
-  .factory('ProjectsService', [
-    '$location',
-    '$q',
-    '$routeParams',
-    'AuthService',
-    'DataService',
+  .factory('ProjectsService',
     function($location, $q, $routeParams, AuthService, DataService) {
+
+      // eliminates empty Annotations
+      // could potentially move this into a shared Util service 
+      var cleanAnnotations = function(resource) {
+          resource
+            .metadata
+            .annotations = _.reduce(
+                            resource.metadata.annotations,
+                            function(memo, val, key) {
+                              if(val) {
+                                memo[key] = val;
+                              }
+                              return memo;
+                            }, {});
+          return resource;
+      };
```

Comment:
Editing any other annotations on a namespace would cause an authorization
failure
On Jan 15, 2016 3:50 PM, "Sam Padgett" <notifications@github.com> wrote:

> In assets/app/scripts/services/projects.js
> <https://github.com/openshift/origin/pull/6667#discussion_r49903434>:
>
> > +
> > +      // eliminates empty Annotations
> > +      // could potentially move this into a shared Util service
> > +      var cleanAnnotations = function(resource) {
> > +          resource
> > +            .metadata
> > +            .annotations = _.reduce(
> > +                            resource.metadata.annotations,
> > +                            function(memo, val, key) {
> > +                              if(val) {
> > +                                memo[key] = val;
> > +                              }
> > +                              return memo;
> > +                            }, {});
> > +          return resource;
> > +      };
>
> Hm, I'm a little worried about potentially removing other empty
> annotations we don't know about. I'd suggest only removing display name and
> description to be safe.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6667/files#r49903434>.
>